### PR TITLE
Fix ENTER on macos

### PR DESCRIPTION
**Pull request by fleytman**
After merge https://github.com/magmax/python-readchar/pull/79 in macos was broken enter. It's fix.

And i include https://github.com/magmax/python-readchar/pull/70

### CODE REVIEW

File: readchar/__init__.py [hunk 1]
```diff
@@ -6,11 +6,7 @@
 from sys import platform
 
 
-if (
-    platform.startswith("linux")
-    or platform == "darwin"
-    or platform.startswith("freebsd")
-):
+if platform.startswith(("linux", "darwin", "freebsd")):
```
Comment (Cube707) [inline readchar/__init__.py:9]: thats a cool idea. very nice.

File: readchar/_macos_key.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# common
+
```
Comment (Cube707) [inline readchar/_macos_key.py:2]: please import exsiting definitions from other files here
```suggestion
from ._posix_key import *
```

File: readchar/_macos_key.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+CR = "\x0d"
```
Comment (Cube707) [inline readchar/_macos_key.py:3]: this is alread defined. no need to redefine it

File: readchar/_posix_read.py [hunk 4]
```diff
@@ -19,7 +19,10 @@ def readchar() -> str:
         term[3] &= ~(termios.ICANON | termios.ECHO | termios.IGNBRK | termios.BRKINT)
         termios.tcsetattr(fd, termios.TCSAFLUSH, term)
 
-        ch = sys.stdin.read(1)
+        if sys.platform == "darwin":
+            ch = sys.stdin.readline(1)
+        else:
+            ch = sys.stdin.read(1)
```
Comment (Cube707) [inline readchar/_posix_read.py:25]: this is defnetly to complex for me to just sign of on. Please remove #70 from this PR. I will decide if it makes sense to include it once I finde the time to get around to it.

File: readchar/key.py [hunk 5]
```diff
@@ -4,12 +4,11 @@
...
 
-if (
-    platform.startswith("linux")
-    or platform == "darwin"
-    or platform.startswith("freebsd")
-):
+if platform.startswith(("linux", "freebsd")):
     from ._posix_key import *
+elif platform.startswith("darwin"):
+    from ._posix_key import *
```
Comment (Cube707) [inline readchar/key.py:10]: like I said above, this should be part of the `_mac_key.py` file
```suggestion
```

**Review by Cube707 — CHANGES_REQUESTED** [review]
First of all, thanks for contributing, but your PR is not well formated.

Please keep your commits atomic (meaning each one should containe a single standalone change), so avoid commit that just fix a small mistake in a previos commit (like your last commit that just adds a missing `\n`). Please create a first commit for your refractoring of the `if` statements and than a second commit with your macos-fix.

Also your commit messages contain some random text. keep the first line short and descriptive and use the second line to give further explanations if necessary ([see herer for more info](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)). And only link other issues / PR's on commits that specifically fix them, not when the fix is mixed with other stuff.

-----

Please remove #70 from this PR. I will take care of it once I have some time for it, but it is to complex and I don't know enought about macos to just handle it quick and dirty. For now this will have to stay a limitation (which is fine as MacOS is not supported anyway)

**Comment by Cube707** [comment]
I am not jet sure if I prefere the "proper" way of having a seperate file for mac, or if I am fine with a dirty override in the `__init__` file (nobody needs a file with only one line...). I will have to sleep on that. leav it as is for now

**Comment by Cube707** [comment]
also please rebase to current master, as some stuff changed in 77c1ac5

**Comment by Cube707** [comment]
and do you have access to a mac? can you provide screenshot of the `tests/manual-test.py` file not working currently with ENTER but after the fix? That would be great for documentation.

**Comment by fleytman** [comment]
Now my Mac uses \n, but I remember exactly what didn't work and was saved only by changing to \r. I think now you can close the MP, and if I reproduce there will already be additional data in what situation such a problem appears.

**Comment by Cube707** [comment]
OK, thats weird 🤦🏼‍♂️ 
Definitely open a new issue if you can reproduce it 👍🏼

**Comment by Cube707** [comment]
you can still open a PR for your refactoring of the `if` statements. Thats a nice idea.

**Pull request closed without merging**